### PR TITLE
Extract the login form into its own directive

### PR DESCRIPTION
**Pull request by robertknight**
This makes the login form consistent with other recently
written components by having the controller and directive in a single
file and using an element directive in the main app.

Having all the logic in one place should make it a little easier when we come to move the login form to a popup to avoid problems with third-party cookie restrictions and enable OAuth login flows in future.

### CODE REVIEW

File: h/static/scripts/directive/login-form.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      controller: Controller,
+      controllerAs: 'vm',
+      restrict: 'E',
+      scope: {
+        onClose: '&',
+      },
+      templateUrl: 'login_form.html',
+    };
+  },
+  controller: Controller,
```
Comment (nickstenning) [inline h/static/scripts/directive/login-form.js]: The naming here should be consistent. If the thing is a constructor, it should always be `Controller`. If it's not, it should always be `controller`.

**Comment by nickstenning** [comment]
I think you also want to commit the removal of `static/scripts/auth-controller.js` here.

**Comment by robertknight** [comment]
> I think you also want to commit the removal of static/scripts/auth-controller.js here.

Done

**Comment by nickstenning** [comment]
LGTM.

**Pull request merged**